packages/@aws-cdk/aws-eks-v2-alpha/lib/cluster.ts

```diff
@@ -519,10 +519,71 @@ export class EndpointAccess {
   }
 }
 
+/**
+ * Options for configuring EKS Auto Mode compute settings.
+ * When enabled, EKS will automatically manage compute resources like node groups and Fargate profiles.
+ */
+export interface ComputeConfig {
+  /**
+   * Names of nodePools to include in Auto Mode.
+   * You cannot modify the built in system and general-purpose node pools. You can only enable or disable them.
+   * Node pool values are case-sensitive and must be `general-purpose` and/or `system`.
+   *
+   * @see - https://docs.aws.amazon.com/eks/latest/userguide/create-node-pool.html
+   *
+   * @default - ['system', 'general-purpose']
+   */
+  readonly nodePools?: string[];
+
+  /**
+   * IAM role for the nodePools.
+   *
+   * @see - https://docs.aws.amazon.com/eks/latest/userguide/create-node-role.html
+   *
+   * @default - generated by the CDK
+   */
+  readonly nodeRole?: iam.IRole;
+
+}
+
+/**
+ * Options for configuring EKS Auto Mode elastic load balancing settings.
+ * When enabled, EKS will automatically manage load balancers.
+ */
+export interface ElasticLoadBalancingConfig {
+  // TODO - implement other props in a seperate PR
+}
+
+/**
+ * Options for configuring EKS Auto Mode Kubernetes networking settings.
+ * Controls automatic management of networking resources in the cluster.
+ */
+export interface KubernetesNetworkConfig {
+  /**
+   * Elastic load balancing configuration for Auto Mode.
+   * Controls how EKS automatically manages load balancers for services.
+   * @default - Auto Mode elastic load balancing disabled
+   */
+  readonly elasticLoadBalancing?: ElasticLoadBalancingConfig;
+}
+
 /**
  * Properties for configuring a standard EKS cluster (non-Fargate)
  */
 export interface ClusterProps extends ClusterCommonOptions {
+  /**
+   * Configuration for compute settings in Auto Mode.
+   * When enabled, EKS will automatically manage compute resources.
+   * @default - Auto Mode compute disabled
+   */
+  readonly compute?: ComputeConfig;
+
+  /**
+   * Configuration for Kubernetes networking settings in Auto Mode.
+   * When enabled, EKS will automatically manage networking resources.
+   * @default - Auto Mode networking disabled
+   */
+  readonly kubernetesNetwork?: KubernetesNetworkConfig;
```

Comment:
`IpFamily` and `ServiceIpv4Cidr` are currently flattened properties. In the L1 resource, they are in KubernetesNetworkConfig. I think we should be consistent here. Either moving them to `KubernetesNetworkConfig` or making `ElasticLoadBalancingConfig` a direct property.

Reply:
Agree. There's no additional props other than `enabled`. So I think it's fine not define the interface until it's necessary in the future.